### PR TITLE
CXP-1361: Locale : Disable catalog if at least 1 FILTER becomes invalid

### PR DESCRIPTION
**Pull request by hogren**
**Description (for Contributor and Core Developer)**
image

image

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/tests/Integration/Infrastructure/Job/DisableCatalogsOnLocaleDeactivationConstraintTest.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        $this->validator = self::getContainer()->get(ValidatorInterface::class);
+        $this->disableCatalogConstraint = self::getContainer()->get(DisableCatalogsOnLocaleDeactivationConstraint::class);
+
+        $this->purgeData();
+    }
+
+    public function testItProvidesConstraintToValidateParameters(): void
+    {
+        $constraintCollection = $this->disableCatalogConstraint->getConstraintCollection();
+        $parameters = ['locale_codes' => ['EUR']];
```
Comment (eoliveauakeneo) [inline components/catalogs/back/tests/Integration/Infrastructure/Job/DisableCatalogsOnLocaleDeactivationConstraintTest.php:29]: You forgot to change this currency

File: components/catalogs/back/tests/Integration/Infrastructure/Job/DisableCatalogsOnLocaleDeactivationConstraintTest.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            'missing locale codes' => [
+                'parameters' => [],
+                'expectedMessage' => 'This field is missing.',
+            ],
+            'locale codes is not an array' => [
+                'parameters' => [
+                    'locale_codes' => 'fr_FR',
+                ],
+                'expectedMessage' => 'This value should be of type array.',
+            ],
```
Comment (eoliveauakeneo) [inline components/catalogs/back/tests/Integration/Infrastructure/Job/DisableCatalogsOnLocaleDeactivationConstraintTest.php:60]: You can add another tests here like we did on the other PR

`            'channel code is not an string' => [
                'parameters' => [
                    'channel_codes' => ['print', 2, 'mobile'],
                ],
                'expectedMessage' => 'This value should be of type string.',
            ],`

File: components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsUsingLocalesAsFilterQuery.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws \Doctrine\DBAL\Exception
+     * @throws \JsonException
+     */
+    public function execute(array $localeCodes): array
```
Comment (eoliveauakeneo) [inline components/catalogs/back/src/Infrastructure/Persistence/Catalog/GetCatalogIdsUsingLocalesAsFilterQuery.php:27]: I did not found the test associated to this new query on the PR

File: components/catalogs/back/tests/Integration/Infrastructure/Job/DisableCatalogsOnLocaleDeactivationTaskletTest.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        $this->purgeDataAndLoadMinimalCatalog();
+    }
+
+    public function testItDisablesCatalogsOnLocaleDeactivation(): void
+    {
+        $this->getAuthenticatedInternalApiClient();
+
+        $this->createUser('shopifi');
+        $this->createUser('magenta');
+        $this->createCatalog(
```
Comment (RiffVentura) [inline components/catalogs/back/tests/Integration/Infrastructure/Job/DisableCatalogsOnLocaleDeactivationTaskletTest.php:25]: I think you should add another catalog containing `fr_FR` locales so we could check it disables multiple catalogs

**Review by RiffVentura — APPROVED** [review]
Don't forget to update EE fixtures

**Review by eoliveauakeneo — APPROVED** [review]

**Comment by tseho** [comment]
check

**Pull request merged**